openupgrade_scripts/scripts/coupon/15.0.1.0/pre-migration.py

```diff
@@ -5,6 +5,15 @@
 
 @openupgrade.migrate()
 def migrate(env, version):
+    # Delete old template translations before the new template is renamed to ensure that
+    # we remove the terms properly
+    openupgrade.delete_record_translations(
```

Comment:
No. Instead, call the set_xml_ids_noupdate_value() method with value = False.

Reply:
That could be an option, but correct if I'm wrong:

If we set `set_xml_ids_noupdate_value(env, "coupon", ["mail_template_sale_coupon"])`, as the template isn't anymore in the module, it would be unlinked in the `coupon` module update and thus recreated as a new by `sale_coupon` without the annoyance of loading anything from a record.

Reply:
It wouldn't be unlinked in `coupon` update due to the xmlid rename that moves the record to `sale_coupon`. Besides, here in OU we never take the approach of deleting a noupdate template in pre (and wait for its recreation in the module update) for a reason. Instead we always update the template in post with the load_data and then delete its translations.

Reply:
@chienandalu is from today on holidays during 2 weeks, so if you can supersede it, I will appreciate it.

Reply:
After checking correctly the code in both versions, I realized that the `upgrade_analysis_work.txt` was not up to date with the `upgrade_analysis.txt` due to old analysis. So, there was no change in the noupdate situation. Just that the template moved from one module to the other. Also, there were no changes inside the template.

Superseded in https://github.com/OCA/OpenUpgrade/pull/4083.